zivid_samples/src/sample_capture_with_settings_from_file.cpp

```diff
@@ -0,0 +1,81 @@
+#include <ament_index_cpp/get_package_share_directory.hpp>
+#include <exception>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <std_srvs/srv/trigger.hpp>
+
+/*
+ * This sample shows how to set the settings_file_path parameter of the zivid node, subscribe for
+ * the points/xyzrgba topic, and invoke the capture service. When a point cloud is received, a new
+ * capture is triggered.
+ */
+
+void set_settings(const std::shared_ptr<rclcpp::Node> & node)
+{
+  const auto share_directory = ament_index_cpp::get_package_share_directory("zivid_samples");
+  const auto path_to_settings_yml = share_directory + "/settings/camera_settings.yml";
+  RCLCPP_INFO_STREAM(
+    node->get_logger(),
+    "Setting parameter `settings_file_path` to '" << path_to_settings_yml << "'");
+
+  auto param_client = std::make_shared<rclcpp::AsyncParametersClient>(node, "zivid_camera");
+  while (!param_client->wait_for_service(std::chrono::seconds(3))) {
+    if (!rclcpp::ok()) {
+      RCLCPP_ERROR(node->get_logger(), "Client interrupted while waiting for service to appear.");
+      std::terminate();
```

Comment:
Should we replace these terminate's with exceptions?

Reply:
We can do that, I didn't want to replicate the error messages twice, and felt this was simpler than adding another function to every sample. But I'm open to doing that as well.

Reply:
OK, I think we want to use exceptions to follow best-practices and have a clean shutdown, but we don't want to duplicate the text of course. So I think making a small helper function is fine but I would duplicate it so that each sample is standalone.

Reply:
Fixed in 3d9da486dbef8d393f20a2ee589771e73dcecd8c